toke.c

```diff
@@ -7309,18 +7310,18 @@ Perl_yylex(pTHX)
         return yyl_bang(aTHX_ s + 1);
 
     case '<':
+        if (s[1] == '<' && (s == PL_linestart || s[-1] == '\n')
+            && memBEGINs(s+2, (STRLEN) (PL_bufend - (s+2)), "<<<<<"))
```

Comment:
we are  missing the `PL_expect != XOPERATOR` check there?

Reply:
The case where `PL_expect` is `XOPERATOR` (not written with an `else`, but the `if (PL_expect != XOPERATOR)` branch ends with an unconditional `TOKEN()` macro invocation, which does a `return`) had the same code — I deleted four hunks that call `vcs_conflict_marker()`, but added only three.

This change makes sense to me: the location of conflict markers in a file is unrelated to the programmer's intent, and therefore to whether the Perl parser is expecting a term or an operator. So we shouldn't restrict conflict-marker detection to situations with specific parser expectations.

Reply:
thanks for making it clear